sphinx/application.py

```diff
@@ -187,10 +185,13 @@ def __init__(self, srcdir, confdir, outdir, doctreedir, buildername,
 
         # read config
         self.tags = Tags(tags)
-        self.config = Config(self.confdir, CONFIG_FILENAME,
-                             confoverrides or {}, self.tags)
-        self.config.check_unicode()
-        # defer checking types until i18n has been initialized
+        if self.confdir is None:
+            self.config = Config()
```

Comment:
It's possible that the user has provide `confoverrides` but no config file, using a combination of `-C` (no config file) with multiple `-D` (override a general setting) and `-A` (override a `html_context` setting) options. As a result, I think you should pass in `confoverrides` here. I noted this earlier but reproducing here.

```
if self.confdir:
    self.config = Config.read(...)
else:
    self.config = Config({}, confoverrides or {})
```

Reply:
You'll also want to move `check_unicode` outside the if-else block, if you do this.

Reply:
You're right. I have to give `confoverrides` in this case.
Thanks,

Reply:
done